Cit/2882

kpaharelau opened the pull request.

kpaharelau committed:
Receiving hash functionality has been changed according using IPython instead Python
Python calls has been changed to IPython for Python3, Ruby, IPython and Julia languages.

kpaharelau committed:
IPython calls has been changed back to Python for Python3, Ruby, IPython and Julia languages because it not works on Windows OS

kpaharelau committed:
IPython calls has been changed back to Python for Julia language

tomca32 commented on core/src/main/java/com/twosigma/beaker/core/rest/PluginServiceLocatorRest.java:
```diff
@@ -599,11 +600,19 @@ private String hashIPythonPassword(String password, String pluginId, String comm
-    if (null == hash) {
-      throw new RuntimeException("unable to get IPython hash");
+
+    Object[] array = br.lines().filter(new Predicate<String>() {
```
Can this have a bit better name? Calling it "array" is not going to be very helpful to anyone that's going to work with this code later. Maybe calling it something like "hashes" or "linesWithSHA" would be better?

scottdraves committed:
kick travis

kpaharelau committed:
tomca32 commented 16 hours ago
Is it possible that this code slowed down Python initialization? Tests seem to time out on loading a small notebook after 12 seconds. I don't think it's flakiness.

Thanks for a reviewing code

I have tested this code using System.currentTimeMillis()

    long startTime = System.currentTimeMillis();

    Object[] array = br.lines().filter(new Predicate<String>() {
      @Override
      public boolean test(String s) {
        return s.contains("sha1:");
      }
    }).toArray();

    if (array.length == 1) {
      String s = (String) array[0];

      long endTime = System.currentTimeMillis();
      System.out.println("Extracting hash " + (endTime - startTime) + " milliseconds");

      return s.substring(s.indexOf("sha1:"));
    }

    throw new RuntimeException("unable to get IPython hash");

 I received

 Extracting hash 859 milliseconds for Python3 loading
 Extracting hash 860 milliseconds for IRuby loading

 I tested old code

     long startTime = System.currentTimeMillis();
     String hash = br.readLine();
     if (null == hash) {
       throw new RuntimeException("unable to get IPython hash");
     }
     long endTime = System.currentTimeMillis();
     System.out.println("Extracting hash " + (endTime - startTime) + " milliseconds");

     return hash;

I received

 Extracting hash 334 milliseconds for Python3 loading
 Extracting hash 331 milliseconds for IRuby loading

This is not acceptable and code has been refactored

    long startTime = System.currentTimeMillis();
    String line;
    while ((line = br.readLine()) != null) {
      if (line.contains("sha1:")) {

        long endTime = System.currentTimeMillis();
        System.out.println("Extracting hash " + (endTime - startTime) + " milliseconds");

        return line.substring(line.indexOf("sha1:"));
      }
    }
    throw new RuntimeException("unable to get IPython hash");

 I received better results in this case

 Extracting hash 405 milliseconds for Python3 loading
 Extracting hash 349 milliseconds for IRuby loading

kpaharelau committed:
Merge branch 'cit/2882' of https://github.com/twosigma/beaker-notebook into cit/2882

kpaharelau committed:
Try to avoid test timeout

kpaharelau committed:
Merge branch 'master' into cit/2882

kpaharelau committed:
Try to avoid test timeout

kpaharelau committed:
Try to avoid test timeout

kpaharelau committed:
Try to avoid test timeout

kpaharelau committed:
Try to avoid test timeout. Revert to 2Sigma style executing processes. Use python instead IPython for IPython plugin - for te test purposes

kpaharelau committed:
I found something very strange
Problem exists only for the one notebook - closed-section-breaks-table.bkr
In the master we receive error if python uses for receiving hash in this notebook	(edited)
[launcher] Running 1 instances of WebDriver
Started
WARNING - more than one element found for locator By.cssSelector("#open-menuitem") - the first result will be used
[qtp1678046232-24] WARN org.eclipse.jetty.servlet.ServletHandler - /rest/plugin-services/IPython
java.lang.RuntimeException: unable to get IPython hash
    at com.twosigma.beaker.core.rest.PluginServiceLocatorRest.hashIPythonPassword(PluginServiceLocatorRest.java:604)
    at com.twosigma.beaker.core.rest.PluginServiceLocatorRest.generateIPythonConfig(PluginServiceLocatorRest.java:623)
    at com.twosigma.beaker.core.rest.PluginServiceLocatorRest.locatePluginService(PluginServiceLocatorRest.java:417)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.sun.jersey.spi.container.JavaMethodInvokerFactory$1.invoke(JavaMethodInvokerFactory.java:60)
    at com.sun.jersey.server.impl.model.method.dispatch.AbstractResourceMethodDispatchProvider$ResponseOutInvoker._dispatch(AbstractResourceMethodDispatchProvider.java:205)
    at com.sun.jersey.server.impl.model.method.dispatch.ResourceJavaMethodDispatcher.dispatch(ResourceJavaMethodDispatcher.java:75)
    at com.sun.jersey.server.impl.uri.rules.HttpMethodRule.accept(HttpMethodRule.java:302)

​This error exists in the ALL!!! success tests
But we receive timeout exception if ipython uses for a receiving hash.

I have added timeout for the extracting hash method

kpaharelau committed:
timeout now 5 seconds

kpaharelau committed:
timeout now 1 second

kpaharelau committed:
use executor.shutdownNow() instead executor.shutdown()

kpaharelau committed:
Trying to use ProcessBuilder

kpaharelau committed:
Test in the '/beaker-notebook/test/tests/bugs.js' has been commented. Simple code for hash receiving has been restored
